src/java.base/unix/native/criuengine/criuengine.c

```diff
@@ -95,15 +95,26 @@ static int checkpoint(pid_t jvm,
 
     pid_t child = fork();
     if (!child) {
-        const char* args[] = {
+        const char* args[32] = {
             criu,
             "dump",
             "-t", jvmpidchar,
             "-D", imagedir,
             "--shell-job",
             "-v4", "-o", "dump4.log", // -D without -W makes criu cd to image dir for logs
-            NULL
         };
+        const char** arg = args + 10;
+        char *criuopts = getenv("CRAC_CRIU_OPTS");
+        if (criuopts) {
+            char* criuopt = strtok(criuopts, " ");
+            while (criuopt) {
+                *arg++ = criuopt;
+                criuopt = strtok(NULL, " ");
+            }
+        }
+        *arg++ = NULL;
+        assert(ARRAY_SIZE(args) >= (size_t)(arg - args));
```

Comment:
I'm not sure if we are compiling with asserts. Could you turn this and another assert below into a runtime check and explicit failure if arguments do not fit?

Reply:
You're right. We compile product builds with `-DNDEBUG`. I've reworked the code to drop surplus arguments from `CRAC_CRIU_OPTS` and issue a warning if this was necessary.